src/main/java/com/ohmyclass/api/components/subject/entity/Subject.java

```diff
@@ -1,3 +1,42 @@
 package com.ohmyclass.api.components.subject.entity;
 
-public class Subject {}
+import com.fasterxml.jackson.annotation.JsonBackReference;
+import com.fasterxml.jackson.annotation.JsonManagedReference;
+import com.ohmyclass.api.components.group.entity.Group;
+import com.ohmyclass.api.components.task.entity.Task;
+import com.ohmyclass.api.components.user.entity.User;
+import lombok.Getter;
+import lombok.Setter;
+
+import javax.persistence.*;
+import java.util.List;
+import java.util.Set;
+
+@Getter
+@Setter
+@Entity
+@Table(name = "subject")
+public class Subject {
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	@Column(name = "id", nullable = false)
+	private Long id;
+
+	@Column(name = "name")
+	private String name;
+
+	@ManyToMany(fetch = FetchType.EAGER)
+	private List<User> teachers;
```

Comment:
Is it necessary to have multiple teachers per subject?

Reply:
It's not. 
And why is it a reference to "User"? Groups, nor subjects have direct references to User.
Change to "ClassMember".

Reply:
String